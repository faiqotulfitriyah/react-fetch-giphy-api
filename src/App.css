.App {
  text-align: center;
  margin-top: 30px;
}

.App button {
  background-color: transparent;
  color: #05466A;
  font-size: 20px;
  font-style: normal;
  font-weight: 700;
  line-height: normal;
  border: none;
}

h3 {
  margin-left: 10px;
  color: white;
  text-align: left;
}

.button-container {
  margin-bottom: 20px;
}

.trend-gif {
  background-color: black;
  width: 50%;
  margin-top: 40px;
  margin-left: auto;
  margin-right: auto;
}

#trend-gif2 {
  width: 100%;
}

/* .gif-container {
  background-color: black;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: masonry;
  gap: 20px;

} */

.gif-container {
  column-count: 4;
  column-gap: 10px;
}

.gif-container > div {
  margin: 0;
  display: grid;
  grid-template-rows: 1fr auto;
  margin-bottom: 10px;
  break-inside: avoid;
}

.gif-container > div > img {
  grid-row: 1 / -1;
  grid-column: 1;
}

.search-gifs {
  max-width: 50%;
  margin-left: auto;
  margin-right: auto;
}

.gif-container img {
  width: 100%;
  height: auto;
}

.search-container {
  margin: 20px;
}

.search-container input[type="text"] {
  padding: 5px;
  width: 400px;
}

.search-container button {
  padding: 5px 10px;
}

.input-src {
  display: inline-block;
}

.src-input {
  padding-right: 40px;
}

.src-icon-conta {
  position: absolute;
  top: 50%;
  right: 10px;
  transform: translateY(-50%);
  cursor: pointer;
}

.search-box {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}