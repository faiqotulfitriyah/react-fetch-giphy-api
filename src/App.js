import React, { useState } from 'react';
import './App.css';
import TrendGifs from './components/trendingGIF';
import SearchGifs from './components/searchGIF';

export default function App() {
  const [showSearch, setShowSearch] = useState(true);

  const handleTrendClick = () => {
    setShowSearch(false);
  };

  const handleSearchClick = () => {
    setShowSearch(true);
  };

  return (
    <div className="App">
      <button className='trendingBtn' onClick={handleTrendClick}>Trending</button>
      <button className='searchBtn' onClick={handleSearchClick}>Search</button>
      {showSearch ? <SearchGifs /> : <TrendGifs />}
    </div>
  );
}