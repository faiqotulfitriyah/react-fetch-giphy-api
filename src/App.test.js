import React from "react";
import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import App from "./App";
import axios from "axios";

jest.mock("axios");

describe('App Component', () => {

    it('renders correctly', () => {
        const { container } = render(<App />);
        expect(container).toMatchSnapshot();
    });
    
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('switches to TrendGIF component when "Trending" button is clicked', async () => {
        
        render(<App />);       
        const trendButton = screen.getByText('Trending');
        fireEvent.click(trendButton);

        await waitFor(() => {
            const trendComponent = screen.queryByTestId('Trend-GIF');
            expect(trendComponent).toBeTruthy();
        });

        await screen.findByTestId('Trend-GIF');

        const searchComponent = screen.queryByTestId('Search-GIF');
        expect(searchComponent).toBeNull();
    });

    it('switches to SearchGIF component when "Search" button is clicked', async () => {
        render(<App />);

        const searchButton = screen.getByText('Search');
        fireEvent.click(searchButton);

        await waitFor(() => {
            const searchComponent = screen.queryByTestId('Search-GIF');
            expect(searchComponent).toBeTruthy();
        });

        await screen.findByTestId('Search-GIF');
    
        const trendComponent = screen.queryByTestId('Trend-GIF');
        expect(trendComponent).not.toBeNull();
    });

    it('makes an Axios call when "Search" button is clicked', async () => {
        const mockSearchData = {
            data: [
                { id: '3', images: { fixed_height: { url: 'https://example.com/3.gif' } }, title: 'GIF 3' }
            ],
        };

        axios.get.mockResolvedValue({ data: mockSearchData});

        render(<App />);
        const searchInput = screen.getByTestId('search-input');
        const searchButton = screen.getByTestId('search-button');

        fireEvent.change(searchInput, { target: { value: 'test' } });
        fireEvent.click(searchButton);

        await waitFor(() => {
            expect(screen.getByTestId('search-result')).toBeTruthy();
        });
        
        expect(axios.get).toHaveBeenCalledWith('https://api.giphy.com/v1/gifs/search?api_key=CQ4jUWnYsaX3DnV1r4ihtdbHVz74LUF4&q=test');
    });

    it('makes an Axios call when "Trending" button is clicked', async () => {
        const mockTrendData = {
            data: [
                { id: '1', images: { fixed_height: { url: 'https://example.com/1.gif' } }, title: 'GIF 1' },
                { id: '2', images: { fixed_height: { url: 'https://example.com/2.gif' } }, title: 'GIF 2' },
            ]
        }

        axios.get.mockResolvedValue({ data: mockTrendData });

        render(<App />);

        await waitFor(() => {
            expect(screen.getByTestId('Trend-GIF')).toBeTruthy();
        });

        expect(axios.get).toHaveBeenCalledWith('https://api.giphy.com/v1/gifs/trending?api_key=CQ4jUWnYsaX3DnV1r4ihtdbHVz74LUF4')
    });
});