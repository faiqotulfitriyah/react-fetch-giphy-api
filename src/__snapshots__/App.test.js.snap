// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <button
      class="trendingBtn"
    >
      Trending
    </button>
    <button
      class="searchBtn"
    >
      Search
    </button>
    <div
      class="search-gifs"
      data-testid="Search-GIF"
    >
      <div
        class="search-container"
      >
        <form
          class="input-src"
        >
          <div
            class="search-box"
          >
            <input
              class="src-input"
              data-testid="search-input"
              type="text"
              value=""
            />
            <button
              class="src-icon-conta"
              data-testid="search-button"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass btn-search"
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </form>
      </div>
      <div
        class="gif-container"
      />
      <div
        class="trend-gif"
        data-testid="Trend-GIF"
        id="trend-gif2"
      >
        <h3>
          Trending GIFs
        </h3>
        <div
          class="gif-container"
        />
      </div>
    </div>
  </div>
</div>
`;
